### PR TITLE
Upgrade to latest ssb-crypto

### PR DESCRIPTION
**Pull request by pietgeursen**
Breaking changes because it removes Eq / Ord / PartialOrd from a few types but I don't think that _really_ matters.

### CODE REVIEW

File: src/multikey.rs [hunk 1]
```diff
@@ -160,7 +160,7 @@ impl fmt::Display for DecodeLegacyError {
 impl std::error::Error for DecodeLegacyError {}
 
 /// The secret counterpart to Multikey
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Debug, Clone)]
 pub struct Multisecret(SecretKey);
```
Comment (sbillig) [inline src/multikey.rs]: I think this should be `Multisecret(Keypair)`. The libsodium (and thus ssb) "secret key" is actually the public+secret keys, concatenated. The ssb-crypto `SecretKey` type is only half of what you want.

File: src/multikey.rs [hunk 2]
```diff
@@ -173,29 +173,32 @@ impl Multisecret {
         let tail =
             skip_prefix(suffix, ED25519_SUFFIX).ok_or_else(|| DecodeLegacyError::UnknownSuffix)?;
 
-        if data.len() != SECRETKEYBYTES {
```
Comment (sbillig) [inline src/multikey.rs:176]: I think this code was and is broken; the legacy encoding isn't expected to be SECRETKEYBYTES long. I'd say we replace the entire body of this function with a call to `Keypair::from_base64`

File: src/multikey.rs [hunk 3]
```diff
@@ -311,7 +300,7 @@ impl Multisig {
     pub fn to_legacy<W: Write>(&self, w: &mut W) -> Result<(), io::Error> {
         match self.0 {
             _Multisig::Ed25519(ref sig) => {
-                let data = base64::encode_config(&sig[..], base64::STANDARD);
+                let data = base64::encode_config(&sig.0[..], base64::STANDARD);
```
Comment (sbillig) [inline src/multikey.rs]: The base64 encoding could be done by `Signature::as_base64`, `PublicKey::as_base64`, etc

**Review by sbillig — APPROVED** [review]
Thanks @pietgeursen! I've been meaning to do this myself. I added a few comments. Not terribly fussed whether you address them now or not. Merge at will!

File: src/multikey.rs (hunk 1 above)
Comment (pietgeursen) [reply src/multikey.rs]: :open_mouth: 
Nice

File: src/multikey.rs (hunk 2 above)
Comment (pietgeursen) [reply src/multikey.rs:176]: Done

File: src/multikey.rs (hunk 3 above)
Comment (pietgeursen) [reply src/multikey.rs]: Done!

File: src/multikey.rs [hunk 4]
```diff
@@ -173,32 +170,18 @@ impl Multisecret {
...
-                (Multisecret(sk.to_owned()), tail)
-            })
+        Ok((Multisecret(key_pair), tail))
     }
 
     /// Serialize a `Multisecret` into a writer, using the
     /// [legacy encoding](https://spec.scuttlebutt.nz/datatypes.html#multikey-legacy-encoding).
     pub fn to_legacy<W: Write>(&self, w: &mut W) -> Result<(), io::Error> {
-        let data = base64::encode_config(&self.0.0[..], base64::STANDARD);
+        let data = base64::encode_config(&self.0.secret.0, base64::STANDARD);
```
Comment (sbillig) [inline src/multikey.rs]: This one has to use Keypair::as_base64 to match js behavior (base64(pub + sec)).

**Comment by sbillig** [comment]
Btw, feel free to add traits or whatever to ssb-crypto if there's anything you need. I just added the stuff that I had a use for.

File: src/multikey.rs (hunk 4 above)
Comment (pietgeursen) [reply src/multikey.rs]: Done

**Pull request merged**